File: Gabriel_Code/Prototipo_0_Java/App/GraphRegistry.java
package br.pucminas.grafo.core;

import br.pucminas.grafo.graph.AbstractGraph;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;

/**
 * Repositório em memória de instâncias de grafos nomeados.
 * Compartilhado entre todos os handlers via injeção no construtor.
 */
public class GraphRegistry {

    private final Map<String, AbstractGraph> store = new HashMap<>();

    public void register(String id, AbstractGraph graph) {
        store.put(id, graph);
    }

    public AbstractGraph get(String id) {
        AbstractGraph g = store.get(id);
        if (g == null) throw new IllegalArgumentException("Grafo não encontrado: " + id);
        return g;
    }

    public boolean contains(String id) {
        return store.containsKey(id);
    }

    public void remove(String id) {
        store.remove(id);
    }

    public Set<String> listIds() {
        return store.keySet();
    }
}

File: Paulo_Code/Projeto_Python_com_GUI/grafo/core/graph_registry.py
from typing import Dict, Set, Optional

class GraphRegistry:
    def __init__(self):
        self.store = {}

    def register(self, graph_id: str, graph):
        self.store[graph_id] = graph

    def get(self, graph_id: str):
        graph = self.store.get(graph_id)
        if graph is None:
            raise ValueError(f"Grafo não encontrado: {graph_id}")
        return graph

    def contains(self, graph_id: str) -> bool:
        return graph_id in self.store

    def remove(self, graph_id: str):
        if graph_id in self.store:
            del self.store[graph_id]

    def list_ids(self) -> Set[str]:
        return set(self.store.keys())
